src/ReplaySession.cc

```diff
@@ -297,20 +307,21 @@ Completion ReplaySession::enter_syscall(ReplayTask* t,
       // in writeable or shared memory, since in those cases it could be
       // overwritten by the tracee. It could even be dynamically generated and
       // not generated yet.
-      if (t->vm()->is_breakpoint_in_private_read_only_memory(
+      if (!syscall_bp_vm &&
+          t->vm()->is_breakpoint_in_private_read_only_memory(
               syscall_instruction) &&
           t->vm()->add_breakpoint(syscall_instruction, BKPT_INTERNAL)) {
```

Comment:
Don't we need to check here whether syscall_bp_addr and syscall_bp_vm refer to the same vm and address as t/syscall_instruction?